storage_service/locations/api/resources.py

```diff
@@ -259,10 +263,10 @@ class Meta:
 
     def prepend_urls(self):
         return [
+            url(r"^(?P<resource_name>%s)/default/(?P<purpose>[A-Z]{2})%s$" % (self._meta.resource_name, trailing_slash()), self.wrap_view('default'), name="default_location"),
```

Comment:
Here is where we tell tastypie that we want to support a custom endpoint `/location/default/$purpose/`.